libtransmission/peer-msgs.cc

```diff
@@ -577,6 +571,12 @@ class tr_peerMsgsImpl final : public tr_peerMsgs
 
     // ---
 
+    [[nodiscard]] bool can_xfer_metadata() const noexcept
+    {
+        // FIXME(tearfur): do we really need io_->supports_ltep() here? The peer already gave us their metadata id
```

Comment:
supports_ltep() is very lightweight and keeping it here makes the intent clearer, imo.  even if it's redundant with a non-zero id